### PR TITLE
Update the javaagent version to 1.23.0

### LINKED ISSUE #1480: Update the javaagent version to 1.23.1

### PR DESCRIPTION
**Pull request by opentelemetrybot**
Update the javaagent version to `1.23.0`.

cc @laurit @mateuszrzeszutek @trask

Closes #1480

### CODE REVIEW

**Review by trask — APPROVED** [review]

**Review by mateuszrzeszutek — APPROVED** [review]

File: autoinstrumentation/java/version.txt [hunk 1]
```diff
@@ -1 +1 @@
-1.22.1
+1.23.0
```
Comment (frzifus) [inline autoinstrumentation/java/version.txt:1]: Do we need a changelog entry? @open-telemetry/operator-ta-maintainers

**Review by frzifus — APPROVED** [review]

File: autoinstrumentation/java/version.txt (hunk 1 above)
Comment (jaronoff97) [reply autoinstrumentation/java/version.txt:1]: I don't think so, unless this is a breaking version upgrade.

Comment (Aneurysm9) [reply autoinstrumentation/java/version.txt:1]: I think we should include a changelog entry even if it's not breaking as there may be new features added.  Also, we should update to 1.23.1 as there was an issue with 1.23.0.  https://github.com/open-telemetry/opentelemetry-java/releases/tag/v1.23.1

Comment (secustor) [reply autoinstrumentation/java/version.txt:1]: Yes, there should be a entry as we change the current expected behavior.

Comment (frzifus) [reply autoinstrumentation/java/version.txt]: Seems a jar for 1.23.1 is missing.

https://github.com/open-telemetry/opentelemetry-operator/actions/runs/4198078109/jobs/7281231001#step:9:175

Comment (trask) [reply autoinstrumentation/java/version.txt]: There isn't a Javaagent 1.23.1 release. The Javaagent 1.23.0 release was built on top of Java SDK 1.23.1.

Comment (Aneurysm9) [reply autoinstrumentation/java/version.txt]: That's incredibly confusing, though I can understand why.

Comment (pavolloffay) [reply autoinstrumentation/java/version.txt]: The changelog is not needed for this PR. It only publishes new image. 


The image is then specified in the operator in the release PR.

**Comment by pavolloffay** [comment]
Reverting back @frzifus change the 1.23.1 hasn't been cut and the changelog will be added during the release.

**Review by pavolloffay — APPROVED** [review]

File: autoinstrumentation/java/version.txt (hunk 1 above)
Comment (pavolloffay) [reply autoinstrumentation/java/version.txt:1]: See https://github.com/open-telemetry/opentelemetry-operator/pull/1484

**Pull request merged**